### ISSUE TITLE
Update to 2.2.0.4. ends with error

### ISSUE THREAD

**Issue by johny-mnemonic** [mschlenstedt/Loxberry#1182]
Here is the log of the error I am getting when trying to update to latest version 2.2.0.4.
```
 INFO: This will take some time now. We suggest getting a coffee or a second beer :-)
 INFO: Upgrading system kernel and firmware. Takes up to 10 minutes or longer! Be patient and do NOT reboot!
*** Raspberry Pi firmware updater by Hexxeh, enhanced by AndrewS and Dom
*** Performing self-update
*** Relaunching after update
*** Raspberry Pi firmware updater by Hexxeh, enhanced by AndrewS and Dom
*** We're running for the first time
#############################################################
WARNING: 'rpi-update' updates to pre-releases of the linux
kernel tree and Videocore firmware.

'rpi-update' should only be used if there is a specific
reason to do so - for example, a request by a Raspberry Pi
engineer.

DO NOT use 'rpi-update' as part of a regular update process.

##############################################################
*** Downloading specific firmware revision (this will take a few minutes)
% Total % Received % Xferd Average Speed Time Time Time Current
Dload Upload Total Spent Left Speed
0 0 0 0 0 0 0 0 --:--:-- --:--:-- --:--:-- 0 100 168 100 168 0 0 3111 0 --:--:-- --:--:-- --:--:-- 3111
100 585 0 585 0 0 2437 0 --:--:-- --:--:-- --:--:-- 2437 100 9052k 0 9052k 0 0 7626k 0 --:--:-- 0:00:01 --:--:-- 9548k 100 19.9M 0 19.9M 0 0 9323k 0 --:--:-- 0:00:02 --:--:-- 10.2M 100 23.9M 0 23.9M 0 0 6930k 0 --:--:-- 0:00:03 --:--:-- 7434k 100 24.9M 0 24.9M 0 0 4939k 0 --:--:-- 0:00:05 --:--:-- 5179k 100 24.9M 0 24.9M 0 0 4264k 0 --:--:-- 0:00:05 --:--:-- 4443k 100 24.9M 0 24.9M 0 0 3874k 0 --:--:-- 0:00:06 --:--:-- 3050k 100 28.7M 0 28.7M 0 0 4097k 0 --:--:-- 0:00:07 --:--:-- 1813k 100 36.1M 0 36.1M 0 0 4521k 0 --:--:-- 0:00:08 --:--:-- 2688k 100 40.6M 0 40.6M 0 0 3549k 0 --:--:-- 0:00:11 --:--:-- 2453k 100 40.7M 0 40.7M 0 0 1259k 0 --:--:-- 0:00:33 --:--:-- 596k 100 41.0M 0 41.0M 0 0 1266k 0 --:--:-- 0:00:33 --:--:-- 620k 100 48.5M 0 48.5M 0 0 1454k 0 --:--:-- 0:00:34 --:--:-- 750k 100 50.6M 0 50.6M 0 0 1072k 0 --:--:-- 0:00:48 --:--:-- 369k 100 57.8M 0 57.8M 0 0 1204k 0 --:--:-- 0:00:49 --:--:-- 471k 100 63.1M 0 63.1M 0 0 1234k 0 --:--:-- 0:00:52 --:--:-- 1192k 100 67.7M 0 67.7M 0 0 1304k 0 --:--:-- 0:00:53 --:--:-- 1368k 100 70.3M 0 70.3M 0 0 1328k 0 --:--:-- 0:00:54 --:--:-- 1113k 100 71.6M 0 71.6M 0 0 1329k 0 --:--:-- 0:00:55 --:--:-- 3136k 100 72.3M 0 72.3M 0 0 1317k 0 --:--:-- 0:00:56 --:--:-- 2113k 100 72.5M 0 72.5M 0 0 1297k 0 --:--:-- 0:00:57 --:--:-- 1977k 100 72.7M 0 72.7M 0 0 1276k 0 --:--:-- 0:00:58 --:--:-- 985k 100 72.7M 0 72.7M 0 0 1255k 0 --:--:-- 0:00:59 --:--:-- 488k 100 72.8M 0 72.8M 0 0 1238k 0 --:--:-- 0:01:00 --:--:-- 235k 100 72.8M 0 72.8M 0 0 1208k 0 --:--:-- 0:01:01 --:--:-- 92869 100 72.8M 0 72.8M 0 0 1193k 0 --:--:-- 0:01:02 --:--:-- 56173 100 72.8M 0 72.8M 0 0 1129k 0 --:--:-- 0:01:06 --:--:-- 25309 100 72.9M 0 72.9M 0 0 1091k 0 --:--:-- 0:01:08 --:--:-- 18145 100 72.9M 0 72.9M 0 0 1047k 0 --:--:-- 0:01:11 --:--:-- 14721 100 72.9M 0 72.9M 0 0 980k 0 --:--:-- 0:01:16 --:--:-- 11300 100 73.0M 0 73.0M 0 0 942k 0 --:--:-- 0:01:19 --:--:-- 9741 100 73.0M 0 73.0M 0 0 808k 0 --:--:-- 0:01:32 --:--:-- 6208 100 73.0M 0 73.0M 0 0 672k 0 --:--:-- 0:01:51 --:--:-- 3821 100 73.1M 0 73.1M 0 0 666k 0 --:--:-- 0:01:52 --:--:-- 3998 100 73.1M 0 73.1M 0 0 651k 0 --:--:-- 0:01:55 --:--:-- 5479 100 73.2M 0 73.2M 0 0 646k 0 --:--:-- 0:01:55 --:--:-- 5377 100 73.2M 0 73.2M 0 0 599k 0 --:--:-- 0:02:05 --:--:-- 6020 100 73.2M 0 73.2M 0 0 597k 0 --:--:-- 0:02:05 --:--:-- 13769 100 73.2M 0 73.2M 0 0 592k 0 --:--:-- 0:02:06 --:--:-- 12621 100 73.3M 0 73.3M 0 0 587k 0 --:--:-- 0:02:07 --:--:-- 12790 100 73.3M 0 73.3M 0 0 581k 0 --:--:-- 0:02:09 --:--:-- 12394 100 73.3M 0 73.3M 0 0 577k 0 --:--:-- 0:02:10 --:--:-- 32559 100 73.4M 0 73.4M 0 0 573k 0 --:--:-- 0:02:11 --:--:-- 30007 100 73.4M 0 73.4M 0 0 565k 0 --:--:-- 0:02:12 --:--:-- 26010 100 73.4M 0 73.4M 0 0 558k 0 --:--:-- 0:02:14 --:--:-- 23638 100 73.5M 0 73.5M 0 0 552k 0 --:--:-- 0:02:16 --:--:-- 22946 100 73.5M 0 73.5M 0 0 541k 0 --:--:-- 0:02:19 --:--:-- 18143 100 73.5M 0 73.5M 0 0 534k 0 --:--:-- 0:02:20 --:--:-- 16599 100 73.6M 0 73.6M 0 0 522k 0 --:--:-- 0:02:24 --:--:-- 14446 100 73.6M 0 73.6M 0 0 517k 0 --:--:-- 0:02:25 --:--:-- 17648 100 73.7M 0 73.7M 0 0 512k 0 --:--:-- 0:02:27 --:--:-- 17986 100 73.7M 0 73.7M 0 0 507k 0 --:--:-- 0:02:28 --:--:-- 20335 100 73.8M 0 73.8M 0 0 503k 0 --:--:-- 0:02:30 --:--:-- 24800 100 73.8M 0 73.8M 0 0 502k 0 --:--:-- 0:02:30 --:--:-- 38941 100 73.8M 0 73.8M 0 0 465k 0 --:--:-- 0:02:42 --:--:-- 13670 100 74.1M 0 74.1M 0 0 425k 0 --:--:-- 0:02:58 --:--:-- 15757 100 78.3M 0 78.3M 0 0 447k 0 --:--:-- 0:02:59 --:--:-- 156k 100 78.9M 0 78.9M 0 0 448k 0 --:--:-- 0:03:00 --:--:-- 175k 100 79.5M 0 79.5M 0 0 449k 0 --:--:-- 0:03:01 --:--:-- 189k 100 79.9M 0 79.9M 0 0 449k 0 --:--:-- 0:03:02 --:--:-- 316k 100 80.3M 0 80.3M 0 0 448k 0 --:--:-- 0:03:03 --:--:-- 1303k 100 80.8M 0 80.8M 0 0 449k 0 --:--:-- 0:03:04 --:--:-- 518k 100 81.2M 0 81.2M 0 0 449k 0 --:--:-- 0:03:05 --:--:-- 480k 100 81.6M 0 81.6M 0 0 448k 0 --:--:-- 0:03:06 --:--:-- 432k 100 82.0M 0 82.0M 0 0 448k 0 --:--:-- 0:03:07 --:--:-- 436k 100 82.4M 0 82.4M 0 0 448k 0 --:--:-- 0:03:08 --:--:-- 431k 100 82.7M 0 82.7M 0 0 448k 0 --:--:-- 0:03:09 --:--:-- 394k 100 83.1M 0 83.1M 0 0 447k 0 --:--:-- 0:03:10 --:--:-- 388k 100 83.7M 0 83.7M 0 0 448k 0 --:--:-- 0:03:11 --:--:-- 423k 100 84.2M 0 84.2M 0 0 448k 0 --:--:-- 0:03:12 --:--:-- 432k 100 84.3M 0 84.3M 0 0 418k 0 --:--:-- 0:03:26 --:--:-- 107k 100 87.5M 0 87.5M 0 0 425k 0 --:--:-- 0:03:30 --:--:-- 229k 100 88.1M 0 88.1M 0 0 427k 0 --:--:-- 0:03:31 --:--:-- 240k 100 88.3M 0 88.3M 0 0 393k 0 --:--:-- 0:03:50 --:--:-- 123k 100 88.9M 0 88.9M 0 0 395k 0 --:--:-- 0:03:50 --:--:-- 129k 100 89.3M 0 89.3M 0 0 397k 0 --:--:-- 0:03:50 --:--:-- 214k
curl: (18) transfer closed with outstanding read data remaining

gzip: stdin: unexpected end of file
tar: Unexpected EOF in archive
tar: Unexpected EOF in archive
tar: Error is not recoverable: exiting now
 ERROR: Error upgrading kernel and firmware - Error 2
 WARNING: Update file /opt/loxberry/sbin/loxberryupdate/update_v2.0.2.pl requests a reboot of LoxBerry. Please reboot your LoxBerry after the installation has finished.
 ERROR: Update script /opt/loxberry/sbin/loxberryupdate/update_v2.0.2.pl finished with errors.
01:27:13.699 INFO: exec_perl_script /opt/loxberry/sbin/loxberryupdate/update_v2.0.2.pl release=v2.2.0.4 logfilename=/opt/loxberry/log/system/loxberryupdate/20201209_012304_091_update.log cron= updatedir=/tmp/loxberryupdate/mschlenstedt-Loxberry-656de07 with user  - errcode 1
01:27:13.699 ERROR: Update-Script update_v2.0.2 returned errorcode 1. Despite errors loxberryupdate.pl will continue.
01:27:13.699 INFO: Setting update script update_v2.0.2 as failed in general configuration.
01:27:13.700 INFO: Updating update script fail state in general configuration...
01:27:13.702 WARNING: LoxBerry Update was SUCCESSFULL, but skipped at least 1 warnings or errors. Check the log.
01:27:13.703 CRITICAL: LoxBerry Update exited or terminated with an error. Errorcode: 1
```
I already tried multiple times to rule out network issue, but it always ends around the same downloaded size.
There is still around 1GB of free space on the SD card, so it shouldn't be an issues of no free space left either.

**Comment by johny-mnemonic** [comment]
OK. So for anyone who would end on the same issue, the root cause is, that older sdcard images had around 40MB of /boot partition, while new sdcard images have 256MB /boot and rpi-update script seems to be using it as temp storage. And 40MB is obviously not enough for the recent FW/kernel updates...
I managed to resize the /boot partition in-place, but it is not for the faint of heart ;-)

**Comment by mschlenstedt** [comment]
Just for information: This is not the root cause. We use a temp folder in / (root) for rpi-update, because of the small boot partition. But 40 MB is still enough for current kernels (as long as you do not use the 64Bit one).

**Comment by johny-mnemonic** [comment]
@mschlenstedt yes, Loxberry update itself is not using /boot, but `rpi-update` itself is and apparently those 40MB were not enough. 
At least this is what I found on RPI forums, where I also found a script for resizing /boot partition. If I remember correctly, they changed the size of to /boot partition from 40 to 256MB with the upgrade to Buster.

**Comment by christianTF** [comment]
You are wrong. It is as Michael said.
https://github.com/mschlenstedt/Loxberry/blob/68c02955f8c0b5da725db6b07a13c92e70e4b05a/sbin/loxberryupdate/update_v2.0.2.pl#L68

**Comment by johny-mnemonic** [comment]
@christianTF I am sorry, but you probably misread what I wrote.
I wrote, that loxberry update (i.e. loxberryupdate/update_v2.0.2.pl) is NOT using /boot partition. 
What I am saying is, that `/usr/bin/rpi-update` tool is using /boot partition, which is beyond your control, as I guess you are not the authors of that tool. In case you are I apologize ;-)

Also when we are at it. Why are you even using `/usr/bin/rpi-update` tool? Do you need some specific function, which is only available in some bleeding edge kernel/FW version and not in stable Raspbian?

**Comment by christianTF** [comment]
OK, I've misunderstood.
Some plugins use libraries compiled against a specific kernel version. Therefore we control the used kernel in LoxBerry Update.

**Comment by mschlenstedt** [comment]
Maybe I missunderstood you both, but we use option BOOT_PATH=/boot.tmp with rpi-update. As far as I know this option do NOT use original Boot partition. That is why the update still works with LBs small boot partition.

**Comment by johny-mnemonic** [comment]
I see. 
I have honestly no clue what specifying `BOOT_PATH=/boot.tmp ROOT_PATH=/` exactly results in, but in the code of rpi-update, they mention these options only for the purpose of updating SD card that is currently not in use...

In the process of resizing the `/boot` partition I also freed some more space from `/` partition, so when I tried the update after resizing the `/boot` to 256M i had slightly more free space in `/` than previously, but I am pretty sure I monitored remaining free disk space in December when it was first failing on me.

Anyway. I am on latest version without having to resort to clean setup of new Loxberry.
So, thanks for posting here at least now and let's hope future updates will be smooth.
